### PR TITLE
fix: correct destructure and add jsdoc types for plugin

### PR DESCRIPTION
**Pull request by danielroe**

### CODE REVIEW

File: templates/plugin.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
-export default async function({ router, store }) {
+/**
+ * @type {import('@nuxt/types').Plugin}
+ */
+const plugin = async function ({ app: { router }, store }, inject) {
```
Comment (pi0) [inline templates/plugin.js:4]: Maybe keeping app for starter is enough? :see_no_evil:

File: templates/plugin.js [hunk 2]
```diff
@@ -1,3 +1,8 @@
...
   //
 }
+
+export default plugin
```
Comment (pi0) [inline templates/plugin.js:8]: Do we need this for jsdoc?

Comment (danielroe) [reply templates/plugin.js:8]: Yes, unfortunately.

File: templates/plugin.js (hunk 1 above)
Comment (danielroe) [inline templates/plugin.js:4]:
```suggestion
const plugin = async function ({ app, store }, inject) {
```

File: templates/plugin.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
-export default async function({ router, store }) {
+/**
+ * @type {import('@nuxt/types').Plugin}
+ */
+const plugin = async function ({ app, store }, inject) {
   //
 }
+
+export default plugin
```
Comment (danielroe) [inline templates/plugin.js:8]:
```suggestion
/**
 * @type {import('@nuxt/types').Plugin}
 */
export default async function ({ app, store }, inject) {
  //
}
```

File: templates/plugin.js (hunk 2 above)
Comment (danielroe) [reply templates/plugin.js]: Not quite sure why I thought that wouldn't work :thinking: Anyway, fixed.

**Comment by pi0** [comment]
Thanks @danielroe :+1:

**Pull request merged**